### PR TITLE
Fix use of functions

### PR DESCRIPTION
**Pull request by matsl**
## What

Fix use of functions.

## Why

More warnings that can be silenced. Reviewing per commit is probably a good idea since each commit is limited in scope and deals with different things.

## Note

There are some FIXME comments added as well for things that look like silencing the warning might not be the right thing. Have a look if these can be resolved as part of this PR or if we should address them later.

## Before

```
kotl/kexport.el:387:11: Warning: Unused lexical variable `font-lock-auto-fontify'
kotl/kexport.el:388:11: Warning: Unused lexical variable `font-lock-mode-disable-list'
kotl/kexport.el:389:11: Warning: Unused lexical variable `font-lock-mode-enable-list'
kotl/kexport.el:391:11: Warning: Unused lexical variable `hm--html-mode-hook'
kotl/kexport.el:392:11: Warning: Unused lexical variable `psgml-mode-hook'
kotl/klink.el:110:11: Warning: Unused lexical variable `default-dir'
kotl/klink.el:122:10: Warning: Unused lexical variable `default-dir'
kotl/kotl-mode.el:460:26: Warning: global/dynamic var ‘kview’ lacks a prefix
kotl/kotl-mode.el:2778:6: Warning: Use ‘with-current-buffer’ rather than save-excursion+set-buffer
kotl/kview.el:488:13: Warning: assignment to free variable ‘found’
kotl/kview.el:674:27: Warning: global/dynamic var ‘kview’ lacks a prefix
kotl/kview.el:675:29: Warning: global/dynamic var ‘kview’ lacks a prefix
kotl/kview.el:675:29: Warning: Variable ‘kview’ declared after its first use
kotl/kview.el:1016:13: Warning: Unused lexical variable `cell-indent'
hactypes.el:669:10: Warning: Unused lexical variable `Man-notify-method'
hbdata.el:341:44: Warning: the function ‘lmail:to’ is not known to be defined.
hbdata.el:335:59: Warning: the function ‘rmail:to’ is not known to be defined.
hbdata.el:335:39: Warning: the function ‘rmail:summ-msg-to’ is not known to be defined.
hbut.el:1579:16: Warning: Unused lexical variable `types'
hibtypes.el: Warning: the function ‘,’ is not known to be defined.
hibtypes.el:1464:33: Warning: the function ‘display-value’ is not known to be defined.
hibtypes.el:1459:33: Warning: the function ‘display-variable’ is not known to be defined.
hibtypes.el:1453:33: Warning: the function ‘display-boolean’ is not known to be defined.
hmail.el:191:6: Warning: the function ‘rmail:msg-widen’ is not known to be defined.
hmouse-drv.el:865:8: Warning: the function ‘mouse-drag-frame’ is not known to be defined.
hmouse-tag.el:907:19: Warning: Unused lexical variable `jedi:find-file-function'
hmouse-tag.el:1168:10: Warning: Unused lexical variable `find-tag-result'
hrmail.el:348:49: Warning: the function ‘hrmail--highlight-buttons’ is not known to be defined.
hrmail.el:338:47: Warning: the function ‘hrmail--show-msg-and-buttons’ is not known to be defined.
hrmail.el:261:40: Warning: the function ‘hrmail--rmail-forward’ is not known to be defined.
hrmail.el:180:43: Warning: the function ‘hrmail--rmail-cease-edit’ is not known to be defined.
hsettings.el:283:35: Warning: the function ‘hproperty:but-flash’ is not known to be defined.
hsmail.el:72:18: Warning: the function ‘mail+narrow’ is not known to be defined.
hycontrol.el:753:14: Warning: Unused lexical variable `frame-zoom-font-difference'
hycontrol.el:1547:21: Warning: global/dynamic var ‘mode’ lacks a prefix
hycontrol.el:1547:21: Warning: global/dynamic var ‘mode’ lacks a prefix
hui-mini.el:410:11: Warning: Unused lexical variable `hyrolo-add-hook'
hui-mini.el:411:11: Warning: Unused lexical variable `hyrolo-edit-hook'
hui-mouse.el:1087:18: Warning: Unused lexical variable `helm-selection-point'
hui-mouse.el:1463:17: Warning: Unused lexical variable `magit-display-buffer-function'
hui-mouse.el:1486:17: Warning: Unused lexical variable `magit-display-buffer-function'
hui-mouse.el:1501:11: Warning: Unused lexical variable `magit-display-buffer-function'
hui-mouse.el:1507:30: Warning: Unknown slot ‘content’
hui-mouse.el:1508:30: Warning: Unknown slot ‘washer’
hui-mouse.el:1509:37: Warning: Unknown slot ‘start’
hui-mouse.el:1510:30: Warning: Unknown slot ‘hidden’
hui-mouse.el:1547:6: Warning: the function ‘unix-apropos-get-man’ is not known to be defined.
hui-mouse.el:1325:13: Warning: the function ‘lmail:delete’ is not known to be defined.
hui-mouse.el:1324:25: Warning: the function ‘lmail:undelete-all’ is not known to be defined.
hui-mouse.el:1318:18: Warning: the function ‘rmail:msg-prev’ is not known to be defined.
hui-mouse.el:1291:13: Warning: the function ‘lmail:goto’ is not known to be defined.
hui-mouse.el:1290:25: Warning: the function ‘lmail:expunge’ is not known to be defined.
hui-mouse.el:1284:18: Warning: the function ‘rmail:msg-next’ is not known to be defined.
hui-mouse.el:1089:14: Warning: the function ‘helm-get-current-action’ is not known to be defined.
hui-mouse.el:815:8: Warning: the function ‘company-quickhelp-manual-begin’ is not known to be defined.
hui-mouse.el:555:20: Warning: the function ‘symlink-referent’ is not known to be defined.
hui-treemacs.el:25:17: Warning: the function ‘package-activate’ is not known to be defined.
hypb.el:718:20: Warning: the function ‘straight-bug-report-package-info’ is not known to be defined.
hypb.el:461:4: Warning: the function ‘helm-info’ is not known to be defined.
hypb.el:452:4: Warning: the function ‘helm-apropos’ is not known to be defined.
hypb.el:290:4: Warning: the function ‘devdocs-lookup’ is not known to be defined.
hyrolo.el:1111:99: Warning: reference to free variable ‘child’
hyrolo.el:1203:11: Warning: Unused lexical variable `plstore-cache-passphrase-for-symmetric-encryption'
hyrolo.el:1232:10: Warning: Unused lexical variable `helm-org-rifle-show-full-contents'
hyrolo.el:1254:14: Warning: Unused lexical variable `helm-org-rifle-show-full-contents'
hyrolo.el:1534:12: Warning: Unused lexical variable `hproperty:but-emphasize-flag'
hyrolo.el:1294:6: Warning: the function ‘org-roam-db-autosync-mode’ is not known to be defined.
hyrolo.el:1289:24: Warning: the function ‘org-roam’ is not known to be defined.
hyrolo.el:1255:10: Warning: the function ‘helm-org-rifle-org-directory’ is not known to be defined.
hyrolo.el:1238:6: Warning: the function ‘helm-org-rifle-files’ is not known to be defined.
hyrolo.el:308:39: Warning: the function ‘markdown-mode’ is not known to be defined.
```
## After

```
kotl/kexport.el:387:11: Warning: Unused lexical variable `font-lock-auto-fontify'
kotl/kexport.el:388:11: Warning: Unused lexical variable `font-lock-mode-disable-list'
kotl/kexport.el:389:11: Warning: Unused lexical variable `font-lock-mode-enable-list'
kotl/kexport.el:391:11: Warning: Unused lexical variable `hm--html-mode-hook'
kotl/kexport.el:392:11: Warning: Unused lexical variable `psgml-mode-hook'
kotl/klink.el:110:11: Warning: Unused lexical variable `default-dir'
kotl/klink.el:122:10: Warning: Unused lexical variable `default-dir'
kotl/kotl-mode.el:460:26: Warning: global/dynamic var ‘kview’ lacks a prefix
kotl/kview.el:488:13: Warning: assignment to free variable ‘found’
kotl/kview.el:674:27: Warning: global/dynamic var ‘kview’ lacks a prefix
kotl/kview.el:675:29: Warning: global/dynamic var ‘kview’ lacks a prefix
kotl/kview.el:675:29: Warning: Variable ‘kview’ declared after its first use
kotl/kview.el:1016:13: Warning: Unused lexical variable `cell-indent'
hactypes.el:669:10: Warning: Unused lexical variable `Man-notify-method'
hbdata.el:341:44: Warning: the function ‘lmail:to’ is not known to be defined.
hbdata.el:335:59: Warning: the function ‘rmail:to’ is not known to be defined.
hbdata.el:335:39: Warning: the function ‘rmail:summ-msg-to’ is not known to be defined.
hbut.el:1579:16: Warning: Unused lexical variable `types'
hibtypes.el: Warning: the function ‘,’ is not known to be defined.
hibtypes.el:1464:33: Warning: the function ‘display-value’ is not known to be defined.
hibtypes.el:1459:33: Warning: the function ‘display-variable’ is not known to be defined.
hibtypes.el:1453:33: Warning: the function ‘display-boolean’ is not known to be defined.
hmail.el:191:6: Warning: the function ‘rmail:msg-widen’ is not known to be defined.
hmouse-tag.el:1169:10: Warning: Unused lexical variable `find-tag-result'
hrmail.el:348:49: Warning: the function ‘hrmail--highlight-buttons’ is not known to be defined.
hrmail.el:338:47: Warning: the function ‘hrmail--show-msg-and-buttons’ is not known to be defined.
hrmail.el:261:40: Warning: the function ‘hrmail--rmail-forward’ is not known to be defined.
hrmail.el:180:43: Warning: the function ‘hrmail--rmail-cease-edit’ is not known to be defined.
hycontrol.el:753:14: Warning: Unused lexical variable `frame-zoom-font-difference'
hycontrol.el:1547:21: Warning: global/dynamic var ‘mode’ lacks a prefix
hycontrol.el:1547:21: Warning: global/dynamic var ‘mode’ lacks a prefix
hui-mini.el:410:11: Warning: Unused lexical variable `hyrolo-add-hook'
hui-mini.el:411:11: Warning: Unused lexical variable `hyrolo-edit-hook'
hui-mouse.el:1085:18: Warning: Unused lexical variable `helm-selection-point'
hui-mouse.el:1505:30: Warning: Unknown slot ‘content’
hui-mouse.el:1506:30: Warning: Unknown slot ‘washer’
hui-mouse.el:1507:37: Warning: Unknown slot ‘start’
hui-mouse.el:1508:30: Warning: Unknown slot ‘hidden’
hui-mouse.el:1545:6: Warning: the function ‘unix-apropos-get-man’ is not known to be defined.
hui-mouse.el:1323:13: Warning: the function ‘lmail:delete’ is not known to be defined.
hui-mouse.el:1322:25: Warning: the function ‘lmail:undelete-all’ is not known to be defined.
hui-mouse.el:1316:18: Warning: the function ‘rmail:msg-prev’ is not known to be defined.
hui-mouse.el:1289:13: Warning: the function ‘lmail:goto’ is not known to be defined.
hui-mouse.el:1288:25: Warning: the function ‘lmail:expunge’ is not known to be defined.
hui-mouse.el:1282:18: Warning: the function ‘rmail:msg-next’ is not known to be defined.
hui-mouse.el:553:20: Warning: the function ‘symlink-referent’ is not known to be defined.
hui-treemacs.el:25:17: Warning: the function ‘package-activate’ is not known to be defined.
hyrolo.el:1115:99: Warning: reference to free variable ‘child’
hyrolo.el:1207:11: Warning: Unused lexical variable `plstore-cache-passphrase-for-symmetric-encryption'
hyrolo.el:1236:10: Warning: Unused lexical variable `helm-org-rifle-show-full-contents'
hyrolo.el:1258:14: Warning: Unused lexical variable `helm-org-rifle-show-full-contents'
hyrolo.el:1538:12: Warning: Unused lexical variable `hproperty:but-emphasize-flag'
```

### CODE REVIEW

**Comment by matsl** [comment]
@rswgnu I rebased this after the recent changes so it is easier to review. Maybe you have some time to have a look at it again?

File: hui-mouse.el [hunk 1]
```diff
@@ -139,7 +136,6 @@ Its default value is `smart-scroll-down'.  To disable it, set it to
 
 (declare-function helm-action-window "ext:helm-lib")
 (declare-function helm-buffer-get "ext:helm-lib")
-;; (declare-function helm-get-current-action "ext:helm-?")
```
Comment (rswgnu) [inline hui-mouse.el:142]: I thought you were keeping that uncommented so it would handle my custom helm changes.

File: hui-mouse.el [hunk 2]
```diff
@@ -166,11 +162,12 @@ Its default value is `smart-scroll-down'.  To disable it, set it to
 (declare-function gnus-topic-read-group "gnus-topic")
 
 (declare-function company-show-doc-buffer "ext:company")
-;; (declare-function company-quick-help-manual-begin "ext:company?")
+(declare-function company-quickhelp-manual-begin "ext:company-quickhelp")
 (declare-function company-show-location "ext:company")
 (declare-function company-select-mouse "ext:company")
 
 
+
```
Comment (rswgnu) [inline hui-mouse.el:170]: This should be just one blank line here.

**Review by rswgnu — APPROVED** [review]

File: hui-mouse.el (hunk 1 above)
Comment (matsl) [reply hui-mouse.el:142]: I did not like the question mark in the filename so just removed it. We can come back to `helm-get-current-action` later. Feel free to address it.

**Pull request merged**